backend/src/src-bdpimport/src/main/java/com/bosch/osmi/sw360/bdp/datasink/thrift/ThriftExchange.java

```diff
@@ -0,0 +1,257 @@
+ */
+package com.bosch.osmi.sw360.bdp.datasink.thrift;
+
+import com.siemens.sw360.datahandler.common.CommonUtils;
+import com.siemens.sw360.datahandler.thrift.components.Component;
+import com.siemens.sw360.datahandler.thrift.components.Release;
+import com.siemens.sw360.datahandler.thrift.licenses.License;
+import com.siemens.sw360.datahandler.thrift.projects.Project;
+import com.siemens.sw360.datahandler.thrift.users.User;
+import com.siemens.sw360.datahandler.thrift.vendors.Vendor;
+import org.apache.log4j.Logger;
+import org.apache.thrift.TException;
+
+import java.util.*;
+import java.util.function.Function;
+import java.util.function.Predicate;
+import java.util.stream.Collectors;
+
+import static com.siemens.sw360.datahandler.common.CommonUtils.nullToEmptyList;
+
+public class ThriftExchange {
+
+    private static final Logger logger = Logger.getLogger(ThriftExchange.class);
+    private ThriftApi thriftApi;
+
+    public ThriftExchange(ThriftApi thriftApi) {
+        this.thriftApi = thriftApi;
+    }
+
+    public List<Vendor> getAllVendors() {
+        List<Vendor> allVendors = null;
+        try {
+            allVendors = getVendorClient().getAllVendors();
+        } catch (TException e) {
+            logger.error("Could not fetch Vendor list:" + e);
+        }
+        return allVendors;
+    }
+
+    public Project getAccessibleProject(String projectName, User user) throws TException{
+        return nullToEmptyList(getAccessibleProjectsSummary(user))
+                .stream()
+                .collect(Collectors.toMap(Project::getName, Function.identity()))
+                .get(projectName);
+    }
+
+    public Project getAccessibleProjectByBdpId(String bdpId, User user) throws TException{
+        return nullToEmptyList(getAccessibleProjectsSummary(user))
+                .stream()
+                .collect(Collectors.toMap(Project::getBdpId, Function.identity()))
+                .get(bdpId);
+    }
+
+    protected List<Project> getAccessibleProjectsSummary(User user) {
+        List<Project> accessibleProjectsSummary = null;
+        try {
+            accessibleProjectsSummary = getProjectClient().getAccessibleProjectsSummary(user);
+        } catch (TException e) {
+            logger.error("Could not fetch Project list for user with email=[" + user.getEmail() + "]:" + e);
+        }
+        return accessibleProjectsSummary;
+    }
+
+    private com.siemens.sw360.datahandler.thrift.projects.ProjectService.Iface getProjectClient() {
+        return thriftApi.getProjectClient();
+    }
+
+    public Optional<List<Release>> searchReleaseByNameAndVersion(String name, String version) {
+        List<Release> releases = null;
+        try {
+            releases = thriftApi.getComponentClient().searchReleaseByName(name);
+        } catch (TException e) {
+            logger.error("Could not fetch Release list for name=[" + name + "], version=[" + version + "]:" + e);
+        }
+
+        if (!Objects.isNull(releases)) {
```

Comment:
>This method exists to be used as a Predicate, filter(Objects::isNull)
For my taste, `releases != null` is marginally more natural as a condition vs. `!Objects.isNull(releases)`
Just sayin'...